tests-functional/tests/test_contact_request.py

```diff
@@ -0,0 +1,99 @@
+from time import sleep
+from uuid import uuid4
+import pytest
+from test_cases import OneToOneMessageTestCase
+from constants import DEFAULT_DISPLAY_NAME
+from clients.signals import SignalType
+
+@pytest.mark.rpc
+class TestContactRequests(OneToOneMessageTestCase):
+
+
+    @pytest.mark.dependency(name="test_contact_request_baseline")
+    def test_contact_request_baseline(self, contact_request_count=1):
+
+        await_signals = [
+            SignalType.MESSAGES_NEW.value,
+            SignalType.MESSAGE_DELIVERED.value,
+        ]
+
+        for i in range(contact_request_count):
+
+            message_text = f"test_contact_request_{i+1}_{uuid4()}"
+
+            sender = self.initialize_backend(await_signals=await_signals)
+            receiver  = self.initialize_backend(await_signals=await_signals)
+
+            pk_sender = sender.get_pubkey(DEFAULT_DISPLAY_NAME)
+            pk_receiver = receiver.get_pubkey(DEFAULT_DISPLAY_NAME)
+
+            existing_contacts = receiver.get_accounts()
+
+            response = sender.send_contact_request([{"id": pk_receiver, "message": message_text}])
+
+            messages_new_event = receiver.find_signal_containing_pattern(SignalType.MESSAGES_NEW.value, event_pattern=message_text, timeout=60)
+
+            signal_texts = []
+            if "messages" in messages_new_event.get("event", {}):
+                signal_texts.extend(
+                    message["text"]
+                    for message in messages_new_event["event"]["messages"]
+                    if "text" in message
+                )
+            if "chats" in messages_new_event.get("event", {}) and messages_new_event["event"]["chats"]:
+                last_message = messages_new_event["event"]["chats"][0].get("lastMessage", {})
+                if "text" in last_message:
+                    signal_texts.append(last_message["text"])
```

Comment:
I couldn't udnerstand what this part is needed for 🤔 
I think we should just wait for `message.new` with certain `message_id`, then quickly compare the sender and message text.

Reply:
`events/messages` is the most important thing you should rely on here.
`chats` and `activityCenterNotifications` are there basically just to have less signals. We want to notify the app of the related occurred changes in 1 signal. 

So you should check `events/messages` to ensure the message was received,
And you CAN check `chats` and `activityCenterNotifications` if you want to ensure correct status-go behaviour on message reception.